### DISCUSSION TITLE
2022 Roadmap

### DISCUSSION THREAD

**Discussion by ferferga** [jellyfin/jellyfin-vue#1679] [General]
This is a short-term roadmap made to summarise some of the ideas and directions for the client  that were thrown out in Matrix and discussions, in an attempt to have some foresight for what to expect this year.

The [old roadmap](https://github.com/jellyfin/jellyfin-vue/discussions/278) is still applicable, as it contains a comprenhensive list of all the features a fully-fledged Jellyfin client should have.

## Media type redesign
We have been talking about redesigning two key concepts:
* Navigation
* Media consumption UX

### Navigation
Hamburger menus are no longer the trend, as they conflict with many gestures used in mobile devices. The hamburguer menu also feels dated in desktop and it's not really useful for a lot of user flows

### Media consumption UX
All Jellyfin clients play all kind of media: movies, tv shows, music, books, photos... All of the clients serve the purpose of doing this "well", but ideally, we should aim at "great": when listening to music, you should feel you're in a music-specific app, like Spotify. When watching movies and TV, you should feel like you're in Netflix. When reading books, you should feel like you're in Kindle.

Right now we have the same issue as web in that regard: everything it's "too" generic and you don't feel like you're in any of those apps.
This is the key concept of the navigation flow redesign: everything should make you aware you're in a client tailored exclusively for whatever you're consuming at that moment

*[See the Matrix discussion directly for more details about this](https://matrix.to/#/!kPTHLknOmRaexLxUwT:matrix.org/$E-wu2BebxfgIcyYjuHA9UHwSvkSTdPzxBBV7ckvMuGc?via=bonifacelabs.ca&via=t2bot.io&via=matrix.org)* 

These changes are rather complex, so we will use the **[agile development lifecycle](https://en.wikipedia.org/wiki/Agile_software_development)** for it and stick to this process to some degree:

1. Open a PR with real changes to an specific part of the two key principles mentioned above: I don't think concepts are worth it if the proposed changes can't be fully tested or asessed by others. I can make any concept, but I for sure won't be able to cover all user cases or flows.
2. PRs receives feedback and contributions from everybody
3. PR is merged to ``dev`` branch
4. Steps 1 to 3 are repeated. Improvements/fixes can also happen in this period. Commits to master can also be rebased to that branch
5. Once most of the goals that the two key areas want to fulfill are accomplished, a PR from ``dev`` to ``master`` is done polishing all the remnants and bringing everything together

#### Related PRs
https://github.com/jellyfin/jellyfin-vue/pull/1677

## Nuxt and Vue 3
We were hesitating towards dropping Nuxt for Vue 3, as it constrained us a lot in some options (like accessing the root ``App.vue`` component or further configuring the router). Nuxt 3 exposes that and (at least on paper) gives us full control while still helping us in doing all the configuration boilerplate (like managing the router and pages).

## Delivering the client in the main repo
So it can be downloaded using your favourite package manager, in the same way as jellyfin-web, allowing you to run both altogether.
Reecently, some cool stuff related to this happened at https://github.com/jellyfin/jellyfin-vue/pull/1672

**Comment by utkvishwas** [comment]
The biggest feedback or feature request i can give is that the ability to search for subs on the player.
